tests/modal.ts

```diff
@@ -0,0 +1,1097 @@
+        el.hide();
+        expect(el.open).to.equal(false);
+    });
+
+    it('attach-parent', async () => {
+        const root = await fixture (`
+                <div id="parent_el">
+                    <aalam-modal>
+                        <div slot="modal-body"></div>
+                    </aalam-modal>
+                </div>
+        `);
+        const body = document.body;
+        let el = root.querySelector("aalam-modal");
+        el.animationDur = 0;
+        const parent_el = el.parentElement;
+        el.show();
+        expect(el.parentElement).to.equal(body);
+        el.hide();
+        expect(el.parentElement).to.equal(parent_el);
+    });
+
+    it('attach-body', async () => {
+        const body_el = document.body;
+        const el1 = await fixture (html`
+                <aalam-modal>
+                    <div slot="modal-body">
+                        <div style="height:20px;width:30px;background:grey">
+                        </div>
+                        <div style="height:40px;width:20px"></div>
+                    </div>
+                </aalam-modal>`);
+        el1.show();
+        const parent_el = el1.parentElement;
+        expect(body_el).to.equal(parent_el)
+    });
+
+    it('stack', async () => {
+        const el1 = await fixture (html`
+            <div id = "parent_holder">
+                <aalam-modal id="outer_modal">
+                    <div slot="modal-body">
+                        <div style="height:20px;width:30px;background:grey">
+                        </div>
+                        <div id="child_modal">
+                            <aalam-modal id="inner_modal">
+                                <div slot="modal-body">
+                                    <div style="background:orange"></div>
+                                </div>
+                            </aalam-modal>
+                        </div>
+                    </div>
+                </aalam-modal>
+                <aalam-modal id="stack_modal" stack>
+                    <div slot="modal-body"></div>
+                </aalam-modal>
+            </div>`);
+
+        const out_mod = el1.querySelector("#outer_modal");
+        const out_par = out_mod.parentElement;
+        out_mod.show();
+        expect(out_mod.parentElement).to.equal(document.body);
+        const in_mod = out_mod.querySelector("#inner_modal");
+        const in_par = in_mod.parentElement;
+        out_mod.animationDur = 0;
+        in_mod.show();
+        expect(out_mod.parentElement).to.equal(out_par);
+        expect(in_mod.parentElement).to.equal(document.body);
+
+        const stack_modal = el1.querySelector("#stack_modal");
+        out_mod.show();
+        stack_modal.show();
+        stack_modal.animationDur =0;
+        expect(stack_modal.parentElement).to.equal(out_mod.parentElement);
+        expect(stack_modal.parentElement).to.equal(document.body);
+        expect(out_mod.parentElement).to.equal(document.body);
```

Comment:
In all this, I think check the parent element to make sure the modal is show is not convincing. I think it better to check the modal size and bound and see if its really shown in the view.